shuup/front/basket/objects.py

```diff
@@ -85,12 +86,26 @@ def can_change_quantity(self):
 
 
 class BaseBasket(OrderSource):
-    def __init__(self, request, basket_name="basket"):
+    def __init__(self, request, key=None):
         super(BaseBasket, self).__init__(request.shop)
-        self.basket_name = basket_name
         self.request = request
-        if request:
-            self.ip_address = request.META.get("REMOTE_ADDR")
+        self.ip_address = request.META.get("REMOTE_ADDR")
+
+        # FIXME Somehow, not all requests provide a session attribute during
```

Comment:
It does also work without using a session, the API test suite is using only the URLs to access the basket.